pymongo/asynchronous/mongo_client.py

```diff
@@ -896,6 +896,7 @@ async def target() -> bool:
         # this closure. When the client is freed, stop the executor soon.
         self_ref: Any = weakref.ref(self, executor.close)
         self._kill_cursors_executor = executor
+        self._opened = False
```

Comment:
Is this a new change? Before the async beta merge we didn't set `self._opened` here, only before calling `self._init_background()`.

Reply:
Yeah _opened was added in [PYTHON-4347](https://jira.mongodb.org/browse/PYTHON-4347).

Reply:
Can we remove the `self._opened = False` on line 857 then? Otherwise we assign it False twice for no reason.

Reply:
Best practice is to define instance variable only in `__init__`.

Reply:
Oh right I forgot that's the first definition of it.